src/app/main.c

```diff
@@ -213,40 +212,40 @@ rpmostree_print_gpg_verify_result (OstreeGpgVerifyResult *result)
 
 
 static RpmOstreeCommand *
-lookup_command_of_type (RpmOstreeCommand *commands,
-                        const char *name,
-                        const char *type)
+lookup_command (const char *name)
 {
   RpmOstreeCommand *command = commands;
-  const int is_tty = isatty (1);
-  const char *bold_prefix;
-  const char *bold_suffix;
-
-  bold_prefix = is_tty ? "\x1b[1m" : "";  
-  bold_suffix = is_tty ? "\x1b[0m" : "";
 
   while (command->name)
     {
       if (g_strcmp0 (name, command->name) == 0)
-        {
-          if (type)
-            g_printerr ("%snotice%s: %s is %s command and subject to change.\n",
-                        bold_prefix, bold_suffix, name, type);
-          return command;
-        }
+        return command;
       command++;
     }
   return NULL;
 }
 
 const char *
 rpmostree_subcommand_parse (int *inout_argc,
-                            char **inout_argv)
+                            char **inout_argv,
+                            RpmOstreeCommandInvocation *invocation)
 {
   const int argc = *inout_argc;
   const char *command_name = NULL;
   int in, out;
 
+  if (invocation && (invocation->command->flags & RPM_OSTREE_BUILTIN_FLAG_EXPERIMENTAL) > 0)
+    {
+      const int is_tty = isatty (1);
+      const char *bold_prefix = is_tty ? "\x1b[1m" : "";
+      const char *bold_suffix = is_tty ? "\x1b[0m" : "";
+
+      g_assert (invocation);
+
+      g_printerr ("%snotice%s: %s is an experimental command and subject to change.\n",
+                  bold_prefix, bold_suffix, invocation->command->name);
```

Comment:
I think `invocation->command->name` will always be `ex` here. What we actually want is `command_name` below, right? Alternatively, the commands under `ex` also need to be flagged as `EXPERIMENTAL` but not `ex` itself? Hmm, kinda messy.

> the properties of a command are now consistently represented as flags

That makes sense, though for EXPERIMENTAL, since only `ex` will use it, it might be cleaner to drop that flag and print the notice in `rpmostree_builtin_ex()`?